### ISSUE TITLE
RangeError: execution timeout

### ISSUE THREAD

**Issue by mikeyjoel** [jack-valko/Zabbix-Meraki-Discovery#12]
RangeError: execution timeout

I have tried raising the timeout value to 60 since that's the max I can do in Zabbix.
![Screenshot_2021-11-30_09-28-16](https://user-images.githubusercontent.com/12386094/144065930-82ea91a8-3303-4f9a-a76d-150da24ecf21.png)

The RangeError is still an issue. Let me know if the timeout is supposed to be configured elsewhere.
![Screenshot_2021-11-30_09-29-11](https://user-images.githubusercontent.com/12386094/144066022-fcc137a8-d1f4-458d-b429-21b2760b339d.png)

As for the network, we have 240 networks, 5624 clients.
![Screenshot_2021-11-30_09-29-49](https://user-images.githubusercontent.com/12386094/144066073-08fe6ef1-70b4-4a2d-90e3-eb26d957e145.png)

Our Zabbix Server got upgraded recently to 5.4.7.
![Screenshot_2021-11-30_09-33-36](https://user-images.githubusercontent.com/12386094/144066741-5f9cbd8d-6084-4a0d-b495-f33f6df9a5c1.png)

We are able to contact meraki api via curl with no issues.
![Screenshot_2021-11-30_09-35-44](https://user-images.githubusercontent.com/12386094/144067014-30909834-0bfa-4dcc-b052-249979cad477.png)

**Comment by jack-valko** [comment]
Hi Michael and thanks for reporting this issue.  Congratulations, you have a large meraki organization.  I'm at a loss to explain the timeout issue, even at 60 seconds that's a really long time to talk to the API.  I wonder if this is truly the bottleneck or if something internal reaches an earlier timeout.   Here are a few things to try:

1. Command line curl.  From the server or proxy where the Meraki Cloud host is monitored from, run two curl commands:
```
curl -k -H 'X-Cisco-Meraki-API-Key: <<APIKEY>>' https://api-mp.meraki.com/api/v1/organizations
```
\<\<APIKEY\>\> is your Meraki API Key
From this output you'll learn your organization id, use that in this next curl command.
```
time curl -k -H 'X-Cisco-Meraki-API-Key: <<APIKEY>>' https://api-mp.meraki.com/api/v1/organizations/<<ORGID>>/devices/statuses
```
\<\<ORGID\>\> is your organization id, found in first command

My expectation is you'll get a lot of output.  At the very end you'll see some timestamp information, let me know how many clock seconds the command took to run. 

2. Zabbix server debugging.  This will take some coordination.  If your proxy or server is busy or throwing a lot of messages it can be difficult to extract the debugging output.  Here are the commands to run, you'll need to execute each of them twice to turn up (and then down) a sufficient level of debugging.
```
zabbix_server --runtime-control log_level_increase="lld manager"
```
After you run this command twice, execute the Meraki Device Discovery in the Meraki Cloud host.  After it errors again, run this command twice to turn off debugging.
```
zabbix_server --runtime-control log_level_decrease="lld manager"
```
I'd like to see what your server or proxy is doing when the error occurs, it should be in the debugging output, usually in /var/log/zabbix/zabbix_server.log or /var/log/zabbix/zabbix_proxy.log.

Thanks for your help in tracking this down,
Jack

**Comment by mikeyjoel** [comment]
@jack-valko  Thank you for your prompt response, highly appreciated!

1.
```
real    0m3.947s
user    0m0.018s
sys     0m0.006s
```

2. Here is a complete transcript
`
root@cmk-zabbix1:/# cd  /var/log/zabbix/
root@cmk-zabbix1:/var/log/zabbix# ls
zabbix_agentd.log        zabbix_agentd.log.12.gz  zabbix_agentd.log.5.gz  zabbix_agentd.log.9.gz   zabbix_server.log.11.gz  zabbix_server.log.4.gz  zabbix_server.log.8.gz
zabbix_agentd.log.1      zabbix_agentd.log.2.gz   zabbix_agentd.log.6.gz  zabbix_server.log        zabbix_server.log.12.gz  zabbix_server.log.5.gz  zabbix_server.log.9.gz
zabbix_agentd.log.10.gz  zabbix_agentd.log.3.gz   zabbix_agentd.log.7.gz  zabbix_server.log.1      zabbix_server.log.2.gz   zabbix_server.log.6.gz
zabbix_agentd.log.11.gz  zabbix_agentd.log.4.gz   zabbix_agentd.log.8.gz  zabbix_server.log.10.gz  zabbix_server.log.3.gz   zabbix_server.log.7.gz
root@cmk-zabbix1:/var/log/zabbix# cp zabbix_server.log zabbix_server.log.bk
root@cmk-zabbix1:/var/log/zabbix# rm zabbix_server.log
root@cmk-zabbix1:/var/log/zabbix# zabbix_server --runtime-control log_level_increase="lld manager"
zabbix_server [2507905]: command sent successfully
root@cmk-zabbix1:/var/log/zabbix# zabbix_server --runtime-control log_level_increase="lld manager"
zabbix_server [2507906]: command sent successfully
root@cmk-zabbix1:/var/log/zabbix# zabbix_server --runtime-control log_level_decrease="lld manager"
zabbix_server [2507951]: command sent successfully
root@cmk-zabbix1:/var/log/zabbix# zabbix_server --runtime-control log_level_decrease="lld manager"
zabbix_server [2507953]: command sent successfully
root@cmk-zabbix1:/var/log/zabbix# cp zabbix_server.log zabbix_server.log.rangeerror.log
root@cmk-zabbix1:/var/log/zabbix# cat zabbix_server.log.rangeerror.log 
  2249:20211130:180912.228 cannot send list of active checks to "192.168.82.98": host [LIC-HALCYON] not monitored
  2214:20211130:180917.788 log level has been increased to 4 (debug)
  2251:20211130:180917.871 cannot send list of active checks to "192.168.82.243": host [CMK-FSMS3] not monitored
  2214:20211130:180918.517 End of zbx_ipc_service_recv():2
  2214:20211130:180918.517 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180918.976 Got signal [signal:10(SIGUSR1),sender_pid:2165,sender_uid:112,value_int:7169(0x00001c01)].
  2214:20211130:180918.976 log level has been increased to 5 (trace)
  2214:20211130:180919.520 End of zbx_ipc_service_recv():2
  2214:20211130:180919.521 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180920.521 End of zbx_ipc_service_recv():2
  2214:20211130:180920.521 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180921.523 End of zbx_ipc_service_recv():2
  2214:20211130:180921.523 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180922.524 End of zbx_ipc_service_recv():2
  2214:20211130:180922.524 zbx_setproctitle() title:'lld manager #1 [processed 0 LLD rules, idle 5.006883sec during 5.006933 sec]'
  2214:20211130:180922.524 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180923.525 End of zbx_ipc_service_recv():2
  2214:20211130:180923.525 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180924.525 End of zbx_ipc_service_recv():2
  2214:20211130:180924.525 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180925.527 End of zbx_ipc_service_recv():2
  2214:20211130:180925.527 In zbx_ipc_service_recv() timeout:1
  2253:20211130:180926.313 cannot send list of active checks to "192.168.56.118": host [ASH-FSMSDEV1] not monitored
  2214:20211130:180926.528 End of zbx_ipc_service_recv():2
  2214:20211130:180926.528 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180927.529 End of zbx_ipc_service_recv():2
  2214:20211130:180927.529 zbx_setproctitle() title:'lld manager #1 [processed 0 LLD rules, idle 5.005062sec during 5.005103 sec]'
  2214:20211130:180927.529 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180928.530 End of zbx_ipc_service_recv():2
  2214:20211130:180928.530 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180929.531 End of zbx_ipc_service_recv():2
  2214:20211130:180929.531 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180930.532 End of zbx_ipc_service_recv():2
  2214:20211130:180930.532 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180931.533 End of zbx_ipc_service_recv():2
  2214:20211130:180931.533 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180932.533 End of zbx_ipc_service_recv():2
  2214:20211130:180932.533 zbx_setproctitle() title:'lld manager #1 [processed 0 LLD rules, idle 5.004656sec during 5.004690 sec]'
  2214:20211130:180932.533 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180933.534 End of zbx_ipc_service_recv():2
  2214:20211130:180933.535 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180934.535 End of zbx_ipc_service_recv():2
  2214:20211130:180934.535 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180934.663 In ipc_service_accept()
  2214:20211130:180934.664 In ipc_service_add_client()
  2214:20211130:180934.664 End of ipc_service_add_client() clientid:31125
  2214:20211130:180934.664 End of ipc_service_accept()
  2214:20211130:180934.664 End of zbx_ipc_service_recv():2
  2214:20211130:180934.664 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180934.664 zbx_ipc_service_recv() code:1300 size:0 data:
  2214:20211130:180934.664 End of zbx_ipc_service_recv():1
  2214:20211130:180934.664 In zbx_ipc_client_send() clientid:31125
  2214:20211130:180934.664 End of zbx_ipc_client_send():SUCCEED
  2214:20211130:180934.664 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180934.664 End of zbx_ipc_service_recv():1
  2214:20211130:180934.664 In zbx_ipc_socket_close()
  2214:20211130:180934.664 End of zbx_ipc_socket_close()
  2214:20211130:180934.664 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180935.665 End of zbx_ipc_service_recv():2
  2214:20211130:180935.665 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180936.666 End of zbx_ipc_service_recv():2
  2214:20211130:180936.666 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180937.667 End of zbx_ipc_service_recv():2
  2214:20211130:180937.667 zbx_setproctitle() title:'lld manager #1 [processed 0 LLD rules, idle 5.133617sec during 5.133719 sec]'
  2214:20211130:180937.667 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180938.668 End of zbx_ipc_service_recv():2
  2214:20211130:180938.668 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180939.669 End of zbx_ipc_service_recv():2
  2214:20211130:180939.669 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.670 End of zbx_ipc_service_recv():2
  2214:20211130:180940.670 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.719 zbx_ipc_service_recv() code:1200 size:287 data:72 0d 01 00 00 00 00 00 | 6a 29 00 00 00 00 00 00 | fe 00 00 00 5b 7b 22 7b | 23 46 53 4e 41 4d 45 7d | 22 3a 22 59 3a 22 2c 22 | 7b 23 46 53 54 59 50 45 | 7d 22 3a 22 4e 54 46 53 | 22 2c 22 7b 23 46 53 44 | 52 49 56 45 54 59 50 45 | 7d 22 3a 22 66 69 78 65 | 64 22 7d 2c 7b 22 7b 23 | 46 53 4e 41 4d 45 7d 22 | 3a 22 44 3a 22 2c 22 7b | 23 46 53 54 59 50 45 7d | 22 3a 22 4e 54 46 53 22 | 2c 22 7b 23 46 53 44 52
  2214:20211130:180940.719 End of zbx_ipc_service_recv():1
  2214:20211130:180940.719 In lld_queue_request()
  2214:20211130:180940.719 queuing discovery rule:68978
  2214:20211130:180940.719 End of lld_queue_request()
  2214:20211130:180940.719 In zbx_ipc_client_send() clientid:2
  2214:20211130:180940.719 End of zbx_ipc_client_send():SUCCEED
  2214:20211130:180940.719 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.719 zbx_ipc_service_recv() code:1200 size:125 data:73 0d 01 00 00 00 00 00 | 6a 29 00 00 00 00 00 00 | 5c 00 00 00 5b 7b 22 7b | 23 44 45 56 4e 41 4d 45 | 7d 22 3a 22 30 20 43 3a | 22 7d 2c 7b 22 7b 23 44 | 45 56 4e 41 4d 45 7d 22 | 3a 22 31 20 59 3a 22 7d | 2c 7b 22 7b 23 44 45 56 | 4e 41 4d 45 7d 22 3a 22 | 32 20 44 3a 22 7d 2c 7b | 22 7b 23 44 45 56 4e 41 | 4d 45 7d 22 3a 22 5f 54 | 6f 74 61 6c 22 7d 5d 00 | 33 af a6 61 4c db 4e 2f | 00 00 00 00 00
  2214:20211130:180940.719 End of zbx_ipc_service_recv():1
  2214:20211130:180940.719 In lld_queue_request()
  2214:20211130:180940.719 queuing discovery rule:68979
  2214:20211130:180940.719 End of lld_queue_request()
  2214:20211130:180940.719 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.720 End of zbx_ipc_service_recv():2
  2214:20211130:180940.720 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.720 End of zbx_ipc_service_recv():2
  2214:20211130:180940.720 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.720 End of zbx_ipc_service_recv():2
  2214:20211130:180940.720 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.720 End of zbx_ipc_service_recv():2
  2214:20211130:180940.720 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.720 zbx_ipc_service_recv() code:1200 size:115467 data:74 0d 01 00 00 00 00 00 | 6a 29 00 00 00 00 00 00 | ea c2 01 00 5b 7b 22 7b | 23 53 45 52 56 49 43 45 | 2e 4e 41 4d 45 7d 22 3a | 22 41 4a 52 6f 75 74 65 | 72 22 2c 22 7b 23 53 45 | 52 56 49 43 45 2e 44 49 | 53 50 4c 41 59 4e 41 4d | 45 7d 22 3a 22 41 6c 6c | 4a 6f 79 6e 20 52 6f 75 | 74 65 72 20 53 65 72 76 | 69 63 65 22 2c 22 7b 23 | 53 45 52 56 49 43 45 2e | 44 45 53 43 52 49 50 54 | 49 4f 4e 7d 22 3a 22 52
  2214:20211130:180940.720 End of zbx_ipc_service_recv():1
  2214:20211130:180940.720 In lld_queue_request()
  2214:20211130:180940.720 queuing discovery rule:68980
  2214:20211130:180940.720 End of lld_queue_request()
  2214:20211130:180940.720 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.735 zbx_ipc_service_recv() code:1001 size:0 data:
  2214:20211130:180940.735 End of zbx_ipc_service_recv():1
  2214:20211130:180940.735 In lld_process_result()
  2214:20211130:180940.735 discovery rule:68978 has been processed
  2214:20211130:180940.735 In zbx_ipc_client_send() clientid:2
  2214:20211130:180940.735 End of zbx_ipc_client_send():SUCCEED
  2214:20211130:180940.735 End of lld_process_result()
  2214:20211130:180940.735 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.750 zbx_ipc_service_recv() code:1001 size:0 data:
  2214:20211130:180940.750 End of zbx_ipc_service_recv():1
  2214:20211130:180940.750 In lld_process_result()
  2214:20211130:180940.750 discovery rule:68979 has been processed
  2214:20211130:180940.751 In zbx_ipc_client_send() clientid:2
  2214:20211130:180940.751 End of zbx_ipc_client_send():SUCCEED
  2214:20211130:180940.751 End of lld_process_result()
  2214:20211130:180940.751 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180940.769 zbx_ipc_service_recv() code:1001 size:0 data:
  2214:20211130:180940.769 End of zbx_ipc_service_recv():1
  2214:20211130:180940.769 In lld_process_result()
  2214:20211130:180940.769 discovery rule:68980 has been processed
  2214:20211130:180940.769 End of lld_process_result()
  2214:20211130:180940.769 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180941.106 Got signal [signal:10(SIGUSR1),sender_pid:2165,sender_uid:112,value_int:7170(0x00001c02)].
  2214:20211130:180941.106 log level has been decreased to 4 (debug)
  2214:20211130:180941.770 End of zbx_ipc_service_recv():2
  2214:20211130:180941.771 In zbx_ipc_service_recv() timeout:1
  2214:20211130:180942.250 Got signal [signal:10(SIGUSR1),sender_pid:2165,sender_uid:112,value_int:7170(0x00001c02)].
  2214:20211130:180942.250 log level has been decreased to 3 (warning)
  2248:20211130:180952.087 failed to accept an incoming connection: from 192.168.27.35: reading first byte from connection failed: [104] Connection reset by peer
root@cmk-zabbix1:/var/log/zabbix# 

`

**Comment by jack-valko** [comment]
Hi Michael,

If it takes no more than 4 seconds to fetch your meraki dashboard payload, I can't imagine your server chewing on it for more than 56 seconds.  I gave you the wrong process name to debug, however, sorry about that.  I've added the correct commands below, you still have you run them twice to get the debugging output.  Thanks for your continued assistance,

Jack
```
zabbix_server --runtime-control log_level_increase="lld worker"
zabbix_server --runtime-control log_level_decrease="lld worker"
```
